### DISCUSSION TITLE
Can't sign out a user from the admin console "Sessions" section

### DISCUSSION THREAD

**Discussion by maxime-bc** [keycloak/keycloak#19585] [Q&A]
Hello, 

I am currently using the  ``01-Login`` Node.js sample from the [auth0-express-webapp-sample](https://github.com/auth0-samples/auth0-express-webapp-sample/tree/master/01-Login) repository  (which uses the [express-openid-connect](https://github.com/auth0/express-openid-connect) library) to connect via OpenID to a Keycloak 21.0.2 server and it is working fine.

In `server.js`, i just updated the [config object](https://github.com/auth0-samples/auth0-express-webapp-sample/blob/master/01-Login/server.js#L20) so that the sample works with Keycloak, as follows : 
```javascript
const config = {
  authRequired: false,
  idpLogout: true,
  authorizationParams: {
    response_type: "code",
  },
  clientSecret: "abcde...",
};
```

My ``.env`` file:
```
CLIENT_ID=dummy
ISSUER_BASE_URL=http://localhost:8080/realms/dummy/
SECRET=123456789
PORT=5000
```

In Keycloak, I have set up a realm named `dummy`, and inside this realm, I have created a client named `dummyclient`. I kept the configuration bare minimum and only followed the "Create client" form : I set the "Client authentication" switch to "On" as I want my client private and then set all the URLs.

Now, when I visit the express sample app, I can log-in and see in the "Sessions" section in the Keycloak admin console that my user is logged in : 

![Capture d’écran 2023-07-10 153443](https://github.com/keycloak/keycloak/assets/49453941/8c9b5a46-d5fc-4a44-83b7-41a3a2b11aa5)

My issue is that when I click on the three dots at the end of the line displaying session info, then on "Sign out", the table is emptied but the Keycloak server print the following warning : ``WARN  [org.keycloak.services.managers.AuthenticationManager] (executor-thread-0) Some clients have been not been logged out for user johndoe in dummy realm: dummyclient``. I can confirm that the user was not logged out as I can still navigate to the "Profile" page on the sample app. The same thing happens when I click on the user name and then click on the button "Logout all  sessions". 

If I try to log out of the sample app after this, the Keycloak server will print a new warning : ``[org.keycloak.events] (executor-thread-9) type=LOGOUT_ERROR, realmId=[...], clientId=null, userId=null, ipAddress=[...], error=session_expired, authSessionParentId=[...], authSessionTabId=[...]``, but according to the previous warning, the user was not logged out so the session should not be expired.

I am new to Keycloak, am I missing something in the Keycloak or in the sample app configuration ?

Thanks for any help.

**Comment by rssaini01** [comment]
Hi @maxime-bc, Did you find the solution for this issue? Actually I'm also facing the same issue and not able to understand why this is happening.

**Comment by maxime-bc** [comment]
Hi @rssaini01,  unfortunately I was not able to find a solution yet.

**Comment by maxime-bc** [comment]
I updated to Keycloak 21.1.2 and I still encounter this issue.

**Reply by rssaini01** [reply]
By the way Where your Keycloak is deployed? I mean, Is it on docker or kubernetes cluster or somewhere else?

**Reply by maxime-bc** [reply]
In my development environment, I just run the Keycloak server binary locally. In production, the Keycloak server is deployed in a Docker container running on a VMware virtual machine.

**Reply by maxime-bc** [reply]
I opened an issue #21599

**Comment by maxime-bc** [comment]
For people encountering the same issue, I posted a solution here https://github.com/keycloak/keycloak/issues/21599#issuecomment-1637667239 using [auth0-express-webapp-sample](https://github.com/auth0-samples/auth0-express-webapp-sample/tree/master/01-Login) as a base to add support for backchannel logout. However, backchannel logout support in ``express-openid-connect`` is still in beta.
